File: URI_1021_banknotes_and_coins.js
const lines = ["10.01"];

const banknotes = [ 100, 50, 20, 10, 5, 2 ];
const coins = [ 1.0, 0.5, 0.25, 0.1, 0.05, 0.01 ];

const money = Number(lines[0]);

const SHIFT = 100;
let safeMoney = money * SHIFT;

console.log("NOTAS:");
for (let banknote of banknotes) {
    const safeBanknote = banknote * SHIFT;
    const quantity = Math.floor(safeMoney / safeBanknote);
    safeMoney %= safeBanknote; 
    console.log(`${quantity} nota(s) de R$ ${banknote.toFixed(2)}`);
}

console.log("MOEDAS:");
for (let coin of coins) {
    const safeCoin = coin * SHIFT;
    const quantity = Math.floor(safeMoney / safeCoin);
    safeMoney %= safeCoin;
    console.log(`${quantity} moeda(s) de R$ ${coin.toFixed(2)}`);
}

File: URI_1021_banknotes_and_coins.java
import java.text.DecimalFormat;
import java.util.Scanner;

public class URI_1021_banknotes_and_coins {

	public static void main(String[] args) {
		int[] banknotes = { 100, 50, 20, 10, 5, 2 };
		double[] coins = { 1.0, 0.5, 0.25, 0.1, 0.05, 0.01 };

		DecimalFormat df = new DecimalFormat("0.00");

		Scanner scanner = new Scanner(System.in);
		double money = scanner.nextDouble();
		
		final byte SHIFT = 100;
		int safeMoney = (int) (money * SHIFT);

		System.out.println("NOTAS:");
		for (int banknote : banknotes) {
			int safeBanknote = banknote * SHIFT;
			int quantity = safeMoney / safeBanknote;
			safeMoney %= safeBanknote;
			System.out.println(quantity + " nota(s) de R$ " + df.format(banknote));
		}

		System.out.println("MOEDAS:");
		for (double coin : coins) {
			int safeCoin = (int) (coin * SHIFT);
			int quantity = safeMoney / safeCoin;
			safeMoney %= safeCoin;
			System.out.println(quantity + " moeda(s) de R$ " + df.format(coin));
		}

		scanner.close();
	}
}
